sources/packages/backend/apps/api/src/route-controllers/education-program-offering/education-program-offering.controller.service.ts

```diff
@@ -279,4 +282,57 @@ export class EducationProgramOfferingControllerService {
       ),
     );
   }
+
+  /**
+   * Transformation util for Program Offering.
+   * @param offering
+   * @param hasExistingApplication is the offering linked to any application.
+   * @returns program offering.
+   */
+  async transformToProgramOfferingDTO(
+    offering: EducationProgramOffering,
+    hasExistingApplication?: boolean,
+  ): Promise<EducationProgramOfferingAPIOutDTO> {
+    const validatedOffering = await this.offeringService.validateOfferingById(
+      offering.id,
+    );
+
+    return {
+      id: offering.id,
```

Comment:
I see that the condition `if (!hasExistingApplication)` is removed here. If update offering is called via API for an offering which has existing application, how will it identify that and throw error(as we have a separate API for basic data update)?

Reply:
The UI should not call it. If called the API must throw an error.

Reply:
I saw the exception thrown at service. 👍